src/pages/Admin/AdminUpdateClass/AdminUpdateClass.tsx

```diff
@@ -13,9 +13,45 @@ import {
 import ClassComponent from './ClassComponent';
 import { useModal } from '../../../components/Modal/useModal';
 import { Modal } from '../../../components/Modal/Modal';
+import { ErrorMessageType } from '../../../types';
+import { ERROR_MESSAGES } from '../../../utilities/constants';
+import { getAdminClass } from '../../../api/getAdminClasses';
+import { AddNewClassForm } from './AddNewClass';
+import { getCsrfToken } from '../../../api/getCsrfToken';
+import { Button, PrimaryButton } from '../../../components/Button';
+
+interface ClassComponentDataProps {
+  _id: string;
+  date: string;
+  name: string;
+  subject: string;
+  classworks?: [];
+}
 
 const AdminUpdateClass = () => {
   const { isOpen, toggle } = useModal();
+  const [classes, setClasses] = useState<ClassComponentDataProps[]>();
+  const { id } = useParams();
+
+  const fetchAdminClasses = useCallback(async () => {
+    try {
+      const res = await getAdminClass(id);
+      setClasses(res.data);
+    } catch (error) {
+      const errors = error as ErrorMessageType[];
+      if (errors?.[0]?.msg === ERROR_MESSAGES.unauthorized) {
+        // navigate('/sign-in', { replace: true });
+      }
+    }
+  }, [id]);
+  const handleCloseModal = () => {
```

Comment:
I also added a reset of the modal values after it's closed so when I open the modal again, there'll be no values from what's previously inputted. Does the modal form gets reset after you close the modal?

Reply:
For me the modal form gets reset automatically when I open again it does not display old data, without having to reset it.